src/layout/verticalPanel/allApplicationList.ts

```diff
@@ -0,0 +1,194 @@
+/** Gnome libs imports */
+import * as Clutter from 'clutter';
+import * as GObject from 'gobject';
+import * as Shell from 'shell';
+import { registerGObjectClass } from 'src/utils/gjs';
+import * as St from 'st';
+import { SearchResultEntry } from './searchResultList';
+const ParentalControlsManager = imports.misc.parentalControlsManager;
+
+/** Extension imports */
+const Me = imports.misc.extensionUtils.getCurrentExtension();
+
+@registerGObjectClass
+export class AllApplicationList extends St.BoxLayout {
+    static metaInfo: GObject.MetaInfo = {
+        GTypeName: 'AllApplicationList',
+        Signals: {
+            'result-selected-changed': {
+                param_types: [SearchResultEntry.$gtype],
+                accumulator: 0,
+            },
+        },
+    };
+    disconnectList: Function[] = [];
+    parentalControlsManager;
+    entrySelected: SearchResultEntry | null = null;
+    constructor(searchEntry: St.Entry) {
+        super({
+            style_class: 'search-result-list',
+            vertical: true,
+        });
+        // Note: Clutter typedefs seem to be incorrect. According to the docs `ev` should be a Clutter.KeyEvent, but it actually seems to be a Clutter.Event.
+        const keyId = searchEntry.clutter_text.connect(
+            'key-press-event',
+            this.onKeyPress.bind(this)
+        );
+        this.disconnectList.push(() => searchEntry.disconnect(keyId));
+        this.parentalControlsManager = ParentalControlsManager.getDefault();
+        const parentalId = this.parentalControlsManager.connect(
+            'app-filter-changed',
+            this.updateResults.bind(this)
+        );
+        this.disconnectList.push(() =>
+            this.parentalControlsManager.disconnect(parentalId)
+        );
+
+        this.connect('destroy', () =>
+            this.disconnectList.forEach((disconnectFunction) =>
+                disconnectFunction()
+            )
```

Comment:
I prefer `for const blah in list` instead of the functional approach, but it doesn't really matter either way.